core/baseline-core-v1.0-psd01.md

```diff
@@ -2963,32 +3036,46 @@ BPI Storage  MUST support authenticated naming systems.
 
 *Non-normative examples include, but are not limited to, certificate authorities or a self-certifying PKI namespace.* 
 
+[[R310]](#r310) Testability: A BPI Storage through one or more access endpointa can be listed in the `Services` section of a DID document. Since BPI Subjects can be associated with a DID of a given DID method -- the authenticated naming system --, and since a DID can be resolved to its DID Document including ite 'Services' section and the cryptographic keys associated with the DID capabilities, and since the W3C DID Standard has an existing test suite for DID methods claiming conformance, the requirement is testable.
+
 #### **[R311]**	
 BPI Storage  MUST support a data exchange protocol that allows for large blocks of data to be replicated. A large data block in this document is defined to be larger than 1MB but less than 128MB. 
 
+[[R311]](#r311) Testability: An example of a BPI Stroage that can replicate large data sizes is the open-source [PostgreSQL](https://www.postgresql.org/) data base, and since PostgreSQL has an extensive test suite, the requirement is testable.
+
 #### **[R312]**	
 BPI Storage  MUST support a routing protocol that enables locating data peers and data objects.
 
 *Non-normative examples are [libp2p](#libp2p) or distributed hash tables [[DHT]](#dht).*
 
+[[R312]](#r312) Testability: An example of BPI Storage that allows for locating other data peer and data objects is the open-source protocol [Interplenetary File System (IPFS)](https://github.com/ipfs/kubo) with its complete test suite. Therefore, the requirement is testable.
+
 #### **[R313]**	
 BPI Storage  MUST support a Network Protocol that handles:
 * NAT traversal such as hole punching, port mapping, and relay
 * Multiple transport protocols
 * Encryption, signing, or clear communications
 * Multi-multiplexes such as Multiplex connections, streams, protocols, peers
 
+[[R313]](#r313) Testability: An example of BPI Storage that incorporates a Network Protocol as described in the requirement is the open-source protocol [Interplenetary File System (IPFS)](https://github.com/ipfs/kubo) with its complete test suite, because it utilizes the P2P Network Protocol [libp2p](#libp2p) with its complete test suite that has the required properties. Therefore, the requirement is testable.
+
 #### **[R314]**	
 Fully Persistent BPI Storage  MUST support Generalized Time Stamps.
 
 *Non-normative examples are conflict-free replicated data types [[CRDT](#crdt)] or Interval Tree Clocks [[ITC](#itc)] to ensure eventual data consistency.*
 
+[[R313]](#r313) Testability: An example of a Fully Persistent BPI Stroage that supports Generalized Time Stamps is the [CRDT Extension](https://github.com/supabase/pg_crdt) of the open-source [PostgreSQL](https://www.postgresql.org/) data base, and since PostgreSQL has an extensive test suite as well as the CRDT extension, the requirement is testable.
```

Comment:
```suggestion
[[R313]](#r313) Testability: An example of a Fully Persistent BPI Storage that supports Generalized Time Stamps is the [CRDT Extension](https://github.com/supabase/pg_crdt) of the open-source [PostgreSQL](https://www.postgresql.org/) data base, and since PostgreSQL has an extensive test suite as well as the CRDT extension, the requirement is testable.
```